.github/workflows/aml-nightly.yml

```diff
@@ -0,0 +1,98 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# ---------------------------------------------------------
+
+name: aml-nightly
+
+on:
+  #          ┌───────────── minute (0 - 59)
+  #          │ ┌───────────── hour (0 - 23)
+  #          │ │ ┌───────────── day of the month (1 - 31)
+  #          │ │ │ ┌───────────── month (1 - 12 or JAN-DEC)
+  #          │ │ │ │ ┌───────────── day of the week (0 - 6 or SUN-SAT)                                  
+  #          │ │ │ │ │
+  #          │ │ │ │ │
+  schedule: 
+    - cron: '0 0 * * *'  # basically running everyday at 12AM
+    # cron works with default branch (main) only: # https://github.community/t/on-schedule-per-branch/17525/2
+  
+  push:
+    # because we can't schedule runs for non-main branches,
+    # to ensure we are running the build on the staging branch, we can add push policy for it
+    branches: [staging]
+
+  # enable manual trigger
+  workflow_dispatch:
+    input:
+      tags:
+        description: 'Tags to label this manual run (optional)'
+        default: 'Anything to describe this manual run'
+
+env:
+  CLUSTER_NAME: "nightly-cluster"
+  EXP_NAME: "nightly_tests"
+  RG: "pradjoshi-aml-rg"
+  WS: "pradjoshi-aml-ws"
+  TEST_LOGS_PATH: '"test_logs.log"'
+  PYTEST_EXIT_CODE: "pytest_exit_code.log"
+
+jobs:
+  nightly-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['"python=3.6"']
+        test-group: ['"group_cpu_001"', '"group_cpu_002"', '"group_cpu_003"', '"group_gpu_001"', '"group_gpu_002"', '"group_gpu_003"', '"group_gpu_004"', '"group_gpu_005"', '"group_gpu_006"', '"group_gpu_007"', '"group_gpu_008"', '"group_gpu_009"', '"group_gpu_010"', '"group_spark_001"']
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+      - name: Setup python
+        uses: actions/setup-python@v2
+        with: 
+          python-version: "3.8"
+      - name: Install azureml-core and azure-cli on a GitHub hosted server
+        run: pip install azureml-core azure-cli
+      - name: Log in to Azure
+        uses: azure/login@v1
+        with:
+          creds: ${{secrets.AML_TEST_CREDENTIALS}}
```

Comment:
it would be good to update the README here https://github.com/microsoft/recommenders/tree/main/tests/ci with all the secrets that need to be set, and the resources that we need to have. 

Also, we need to remove all pipelines that are not used anymore

Reply:
Should I remove Beth's code that includes azure pipeline yaml files and python scripts?

Reply:
yes, all this can be removed. We also need to update https://github.com/Microsoft/Recommenders/wiki/Test-Strategy